vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -359,7 +351,7 @@ public void add(Component... components) {
         for (Component component : components) {
             Objects.requireNonNull(component,
                     "Component to add cannot be null");
-            container.appendChild(component.getElement());
+            getElement().appendChild(component.getElement());
```

Comment:
Since we now use `getElement()` instead of a separate container, duplicating logic from `HasComponents` seems no longer necessary.

The `remove`, `removeAll` and `getChildren` overrides can be removed.

`add` and `addComponentAtIndex` can be changed to this:
```java
    @Override
    public void add(Component... components) {
        HasComponents.super.add(components);

        updateVirtualChildNodeIds();
    }

    @Override
    public void addComponentAtIndex(int index, Component component) {
        HasComponents.super.addComponentAtIndex(index, component);

        updateVirtualChildNodeIds();
    }
```

The same should work for Notification.

Reply:
Done. Nice cleanup. Looks like the components have also been overriding the wrong `add` methods. Changed to override `public void add(Collection<Component> components)` and added tests.